tests/buffering.rs

```diff
@@ -352,10 +336,6 @@ fn test_reclaim_disk_space() {
 
     let output_events = test_util::receive_events(out_rx);
 
-    // There was a race caused by a channel in the mock source, and this
-    // check is here to ensure it's really gone.
-    assert_eq!(source_event_counter.load(Ordering::Acquire), num_events);
```

Comment:
I'm pretty sure the potential problem remains here, and this is a meaningful assertion.
We can remove it, of course, but that'd be effectively muting the failure.

Reply:
https://github.com/timberio/vector/pull/2862/files#r444125759 also applies to this.